src/linear_algebra/tensor_product.lean

```diff
@@ -327,6 +327,37 @@ instance : semimodule R (M ⊗[R] N) := tensor_product.semimodule'
 @[simp] lemma tmul_smul (r : R') (x : M) (y : N) : x ⊗ₜ (r • y) = r • (x ⊗ₜ[R] y) :=
 (smul_tmul _ _ _).symm
 
+section
+
+variables {R'' : Type*} [comm_semiring R''] [has_scalar R'' R']
```

Comment:
I'm a little surprised that there are three sets of scalars in play here.

I can think of situations where three sets of scalars are all relevant (`ℤ`, `ℝ`, `ℂ`, say) but it's not clear to me that we need to have results about all three at once.

Would it be easy to add an example where these instances are needed in a comment?

Reply:
I'm currently in the process of trying to merge, so it's hard for me to provide a direct example - but I think the cases that are particularly useful are the cases when the third scalar is equal to either the first or the second scalar. We've seen this three-scalar pattern before, in `linear_map.is_scalar_tower`.

Reply:
Interesting!

Thinking a little more now I suppose a "natural" example where we want to talk about three sets of scalars at once might be if we have two complex representations `V`, `W` of a group `G` and we consider `V ⊗[ℝ] W`. This is naturally a vector space over `ℂ`, and carries an action of `G` and _these two actions commute_. So I guess that does it. (Same goes for `V →ₗ[ℝ] W`.)

Reply:
I just pushed an update to resolve the conflicts.